### DISCUSSION TITLE
Autostart

### DISCUSSION THREAD

**Discussion by tg44** [tg44/heptapod#9] [General]
How deep do we want to configure the autostart/autorun functionality?

I'm thinking of a [launchd](https://stackoverflow.com/a/133425) file which will be inserted to the `~/Library/LaunchAgents/` folder, and let to run the rules automatically daily 1-2x. As a minimal setup.

Do we need a more hardcore feature set around it? Or just an on/off switch is enough which will run the thing daily? Or should I just push things to crontab and give some minimal config options to which hours we want to run this?

**Comment by GBrachetta** [comment]
I think 2x/day should be enough, no? Unless there's some compulsive project creator around there?

**Reply by jgoldhammer** [reply]
I would start simple by using launchd. A good overview is here: https://www.launchd.info

You could wrap the launchd commands to enable or disable the job completely in heptapod commands
![image](https://user-images.githubusercontent.com/3872101/149275307-e807b865-af44-4ceb-9053-7631e521ba24.png)

You could also allow to run and stop the job immediately 
![image](https://user-images.githubusercontent.com/3872101/149275332-fd9e23cb-a29d-46e4-bab4-48edf05d9741.png)

A good starting point could be:
https://github.com/keybase/client/blob/master/go/launchd/launchd.go

Another idea would be to run heptapo more often, but have a kind of change detection to see if a complete scan is necessary. But this is more sophisticated and maybe overkill and errorprone.
